src/Domain/CoordinateMaps/TimeDependent/Translation.cpp

```diff
@@ -96,42 +165,175 @@ std::array<tt::remove_cvref_wrap_t<T>, Dim> Translation<Dim>::frame_velocity(
              << velocity.size()
              << ") does not match the dimension of the translation map (" << Dim
              << ").");
+  // Section for handling overloaded constructor
+  tt::remove_cvref_wrap_t<DataType> radial_function_value(
+      gsl::at(source_coords, 0));
+  if (f_of_r_ == nullptr) {
+    radial_function_value = 1.0;
+  } else {
+    std::array<tt::remove_cvref_wrap_t<DataType>, Dim> distance_to_center{};
+    Scalar<tt::remove_cvref_wrap_t<DataType>> radius(gsl::at(source_coords, 0));
+    get(radius) = 0.0;
+
+    for (size_t i = 0; i < Dim; i++) {
+      gsl::at(distance_to_center, i) =
+          gsl::at(source_coords, i) - gsl::at(center_, i);
+      radius.get() += square(gsl::at(distance_to_center, i));
+    }
+    radius.get() = sqrt(radius.get());
+    radial_function_value = (*f_of_r_)(radius.get());
+  }
   for (size_t i = 0; i < Dim; i++) {
-    gsl::at(result, i) = make_with_value<tt::remove_cvref_wrap_t<T>>(
-        dereference_wrapper(gsl::at(source_coords, i)), velocity[i]);
+    gsl::at(result, i) = velocity[i] * radial_function_value;
   }
   return result;
 }
 
 template <size_t Dim>
-template <typename T>
-tnsr::Ij<tt::remove_cvref_wrap_t<T>, Dim, Frame::NoFrame>
-Translation<Dim>::jacobian(const std::array<T, Dim>& source_coords) const {
-  return identity<Dim>(dereference_wrapper(source_coords[0]));
+template <typename DataType>
+tnsr::Ij<tt::remove_cvref_wrap_t<DataType>, Dim, Frame::NoFrame>
+Translation<Dim>::jacobian(
+    const std::array<DataType, Dim>& source_coords, const double time,
+    const std::unordered_map<
+        std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
+        functions_of_time) const {
+  if (f_of_r_ == nullptr) {
+    return identity<Dim>(dereference_wrapper(source_coords[0]));
+  } else {
+    std::array<tt::remove_cvref_wrap_t<DataType>, Dim> distance_to_center{};
+    Scalar<tt::remove_cvref_wrap_t<DataType>> radius(gsl::at(source_coords, 0));
+    get(radius) = 0.0;
+    for (size_t i = 0; i < Dim; i++) {
+      gsl::at(distance_to_center, i) =
+          gsl::at(source_coords, i) - gsl::at(center_, i);
+      radius.get() += square(gsl::at(distance_to_center, i));
+    }
+    radius.get() = sqrt(radius.get());
+
+    const DataVector function_of_time =
+        functions_of_time.at(f_of_t_name_)->func(time)[0];
+
+    auto result = make_with_value<
+        tnsr::Ij<tt::remove_cvref_wrap_t<DataType>, Dim, Frame::NoFrame>>(
+        dereference_wrapper(source_coords[0]), 0.0);
+    for (size_t i = 0; i < Dim; i++) {
+      for (size_t j = 0; j < Dim; j++) {
+        if constexpr (std::is_same_v<tt::remove_cvref_wrap_t<DataType>,
+                                     double>) {
```

Comment:
You should be able to use the same code for `double` and `DataVector` because the `first_deriv` function has both a `double` and `DataVector` overload.

Reply:
They do, but since I have to compare the different radii to see if L'Hopital's applies or not, I think I have to check the DataType and parse through if it's a `DataVector` since the points might be above or below the threshold.

Reply:
There's actually a function for just this! In `src/Utilities/Containerhelpers.hpp` there is the `get_size` function. For a `double` this just returns 1 but for a `DataVector` it'll give you the size of the vector. Then in the same file there is a function `get_element` which takes the container, and the size you just got and gives you the element. For a `double`, it just returns the double. For a `DataVector`, it'll give you the ith element. So you can have the innermost loop be:

```cpp
for (size_t k=0 ; k<get_size(radius.get()); k++) {
  if (radius.get() > 1.e-13) {
    result.get(i, j) = (*f_of_r_).first_deriv(get_element(radius.get(), k)) *
      gsl::at(function_of_time, i) *
      gsl::at(distance_to_center, j) / get_element(radius.get(), k);
  } else {
    result.get(i, j) = (*f_of_r_).second_deriv(get_element(radius.get(), k)) *
      gsl::at(function_of_time, i) *
      gsl::at(distance_to_center, j);
  }
}
```

and it *should* handle it. I haven't tested this code yet.

Reply:
oh wow, that's super cool, I'll give it a try see if it works!